### PR TITLE
Service selector for single binary deployment fix.

### LINKED ISSUE #7260: [Helm] Issue with loki service in singleBinary
**Describe the bug**
Fresh install through Helm Chart of single binary is not working when going through deffault service.

Currently we have [here](https://github.com/grafana/loki/blob/dcfba366de2fbe036df31e28200df983d4808e67/production/helm/loki/templates/single-binary/service.yaml#L21):

```yaml
  selector:
    {{- include "loki.selectorLabels" . | nindent 4 }}
```

And we **should** have:

```yaml
  selector:
    {{- include "loki.singleBinarySelectorLabels" . | nindent 4 }}
```

Hence, service point to all instances created by Helm Chart, Canary included !

Promtail cannot push logs and Grafana cannot read them

**To Reproduce**
Fresh install the solution with values:

```yaml
loki:
  storage:
      type: 'filesystem'
    rulerConfig:
      storage:
        type: local
```

**Expected behavior**
Included default service should be working

**Environment:**
 - Infrastructure: Kubernetes
 - Deployment tool: helm

**Logs**

```log
level=error ts=2022-09-27T11:18:17.537024173Z caller=client.go:380 component=client host=loki:3100 msg="final error sending batch" status=404 error="server returned HTTP status 404 Not Found (404): 404 page not found"
level=error ts=2022-09-27T11:18:18.337102338Z caller=client.go:380 component=client host=loki:3100 msg="final error sending batch" status=404 error="server returned HTTP status 404 Not Found (404): 404 page not found"
```

### PR DESCRIPTION
**Pull request by besteban1989**
Single binary deployment is not working when going through default service. The issue is with the labels that currently includes the canary pods.

Fixes #7260

### CODE REVIEW

File: production/helm/loki/templates/single-binary/service.yaml [hunk 1]
```diff
@@ -19,5 +19,5 @@ spec:
       targetPort: grpc
       protocol: TCP
   selector:
-    {{- include "loki.selectorLabels" . | nindent 4 }}
-{{- end }}
+    {{- include "loki.singleBinarySelectorLabels" . | nindent 4 }}
```
Comment (jeschkies) [inline production/helm/loki/templates/single-binary/service.yaml:22]: @trevorwhitney unfortunately I don't know enough. This seems fine to me.

Comment (besteban1989) [reply production/helm/loki/templates/single-binary/service.yaml:22]: Any approval for the PR then?

**Review by vlad-diachenko — CHANGES_REQUESTED** [review]
Hey @besteban1989 . 
Huge thanks for the contribution. 
Also, statefulset might add additional labels defined by user in `.Values.singleBinary.selectorLabels` . see: 
https://github.com/grafana/loki/blob/main/production/helm/loki/templates/single-binary/statefulset.yaml#L33-L36

Could you please also add this part here?

**Comment by besteban1989** [comment]
@vlad-diachenko the change is done, pls review and approve if everything is ok.

**Review by vlad-diachenko — APPROVED** [review]
huge thanks @besteban1989 🚀

**Pull request merged**